### PR TITLE
refactor: separate voting period to verification/delivery voting period in DataSale

### PR DESCRIPTION
**Pull request by inchori**
Separate voting period to verification/delivery voting period in DataSale.
- Refactor `abci_test.go` as well.

### CODE REVIEW

File: x/datadeal/abci.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
 				panic("invalid verifiable CID consensus value")
 			}
 
-			keeper.AddDataDeliveryQueue(ctx, dataSale.VerifiableCid, dataSale.DealId, oracleKeeper.GetVotingPeriod(ctx).VotingEndTime)
+			keeper.AddDataDeliveryQueue(ctx, dataSale.VerifiableCid, dataSale.DealId, dataSale.DeliveryVotingPeriod.VotingEndTime)
```
Comment (gyuguen) [inline x/datadeal/abci.go:43]: I think `DeliveryVotingPeriod` set a here.
```go
dataSale.DeliveryVotingPeriod = oracleKeeper.GetVotingPeriod(ctx)
keeper.AddDataDeliveryQueue(ctx, dataSale.VerifiableCid, dataSale.DealId, dataSale.DeliveryVotingPeriod.VotingEndTime)
```

**Review by gyuguen — COMMENTED** [review]
Thank you.
Check my comment please. :)

File: x/datadeal/abci.go (hunk 1 above)
Comment (H4NLee) [reply x/datadeal/abci.go:43]: Good. I was just writing this comments as well. 👍

Comment (inchori) [reply x/datadeal/abci.go:43]: Ah, I missed it. I'll add it! Thanks!

Comment (inchori) [reply x/datadeal/abci.go:44]: Fixed in [73769b8](https://github.com/medibloc/panacea-core/pull/456/commits/73769b88a9b7749a80f28c42c7f28e7ee838d8a9).

File: x/datadeal/abci_test.go [hunk 2]
```diff
@@ -268,10 +276,14 @@ func (suite abciTestSuite) TestDataVerificationEndBlockerVoteRejectSamePower() {
...
 		Status:        types.DATA_SALE_STATUS_VERIFICATION_VOTING_PERIOD,
-		VotingPeriod: &oracletypes.VotingPeriod{
+		VerificationVotingPeriod: &oracletypes.VotingPeriod{
 			VotingStartTime: time.Now().Add(-2 * time.Second),
 			VotingEndTime:   time.Now().Add(-1 * time.Second),
 		},
+		DeliveryVotingPeriod: &oracletypes.VotingPeriod{
+			VotingStartTime: time.Now().Add(1 * time.Second),
+			VotingEndTime:   time.Now().Add(2 * time.Second),
+		},
```
Comment (gyuguen) [inline x/datadeal/abci_test.go:286]: Don't you need this part?
DeliveryVotingPeriod is set a `EndBlocker`

Comment (inchori) [reply x/datadeal/abci_test.go:286]: Yes, the `DeliveryVotingPeriod` set from voting period in `EndBlocker`. However, the test works well if the `DeliveryVotingPeriod` set in `DataVerificationVote`.

Comment (inchori) [reply x/datadeal/abci_test.go]: Actually, the test must be passed if the `DeliveryVotingPeriod` has `nil` value. So I changed the value to `nil` in `DeliveryVotingPeriod` and the tests were passed.

**Review by H4NLee — APPROVED** [review]
lgtm

**Comment by inchori** [comment]
I'll merge this branch into base branch.

**Pull request merged**